packages/next-intl/test/middleware/middleware.test.tsx

```diff
@@ -737,6 +749,48 @@ describe('prefix-based routing', () => {
           'http://localhost:3000/external?hello'
         );
       });
+
+      describe('redirects an internal route with localization to its localized pathname but removes the locale prefix while keeping search params', () => {
```

Comment:
Add tests on paths starting from the current locale, from the default locale, containing two locales and two default locales. next.js had a lot of problems with this in their i18n routing - don't repeat their mistakes!)

Reply:
What am I supposed to do in such scenarios ? `/en/en/internal` => `/en/external` ??
I think this is outside of the scope of this feature, or not sure to understand...

Reply:
Next.js, when you go to the` /en/en/internal` page, will show you the `/en/internal` page as if everything is fine. But formally it is non-existent path and should response with 404 status

It was also funny other time that next.js from this path redirected to /en/internal and showed this page, and after reloading it redirected again - to /internal and again showed as if everything was okay.

Reply:
_Perhaps this is a comment not so much for you as for amannn_

Reply:
@vordgi We had that topic with pathnames that overlap with a locale just recently! 😄 See https://github.com/amannn/next-intl/pull/698.

As far as I know that has been entirely addressed and should no longer be an issue.

Reply:
A-ha-ha, wow, it means you solved this problem much faster than next.js team 😁